### PR TITLE
fix: backpack-demo to use JSON toolbox definitions instead of xml

### PR DESCRIPTION
**Pull request by varshneydevansh**
**Issue**

- #1247 

**Category**

Examples > [backpack-demo](https://github.com/google/blockly-samples/tree/master/examples/backpack-demo)

**Proposed Solution**

Update the code in for `backpack-demo` example `blockly-samples` to use the JSON toolbox API.

**Result**

After testing, it should be working the same way it did before with the XML.

### CODE REVIEW

**Comment by varshneydevansh** [comment]
I think I might need help with JSON as -

image

image


Although, the example at https://google.github.io/blockly-samples/plugins/workspace-backpack/test/index.html seems little different but with my updated JSON, I am able to get the numbers show on the blocks.

**Comment by cpcallen** [comment]
Reassigned to @BeksOmega per instruction from Rachel

File: examples/backpack-demo/index.js [hunk 1]
```diff
@@ -7,22 +7,141 @@
...
+        {
+          "kind": "block",
+          "type": "math_number",
+          "fields": {
+            "NUM": 123
+          }
+        },
+        {
+          "kind": "block",
+          "type": "math_arithmetic"
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: This should include the child shadow blocks =)

File: examples/backpack-demo/index.js [hunk 2]
```diff
@@ -7,22 +7,141 @@
...
+        },
+        {
+          "kind": "block",
+          "type": "math_single"
+        },
+        {
+          "kind": "block",
+          "type": "math_number_property"
+        }
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: These two should also have shadow blocks =)

File: examples/backpack-demo/index.js [hunk 3]
```diff
@@ -7,22 +7,141 @@
...
+          "type": "text_multiline"
+        },
+        {
+          "kind": "block",
+          "type": "text_print"
+        },
+        {
+          "kind": "block",
+          "type": "text_prompt_ext"
+        }
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: These two should also have shadow blocks.

File: examples/backpack-demo/index.js [hunk 4]
```diff
@@ -7,22 +7,141 @@
...
+      "categorystyle": "text_category",
+      "contents": [
+        {
+          "kind": "block",
+          "type": "text"
+        },
+        {
+          "kind": "block",
+          "type": "text_multiline"
+        },
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js:141]: There should be a label here saying "Input/Output:"

**Review by BeksOmega — CHANGES_REQUESTED** [review]
Generally looks good to me! Just needs a few changes to match the currently XML toolbox definition =)

File: examples/backpack-demo/index.js (hunk 4 above)
Comment (varshneydevansh) [reply examples/backpack-demo/index.js:141]: {
          "kind": "label",
          "text": "Input/Output:",
          "web-class": "ioLabel"
        },

        added the lable

File: examples/backpack-demo/index.js (hunk 2 above)
Comment (varshneydevansh) [reply examples/backpack-demo/index.js]: added

File: examples/backpack-demo/index.js [hunk 5]
```diff
@@ -7,22 +7,204 @@
 /**
  * @fileoverview Backpack demo initialization.
  */
+var toolbox = {
+  "kind": "categoryToolbox",
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: Can you change this file to use single quotes?

File: examples/backpack-demo/index.js [hunk 6]
```diff
@@ -7,22 +7,204 @@
...
+    },
+    {
+      "kind": "category",
+      "name": "Loops",
+      "categorystyle": "loop_category",
+      "contents": [
+        {
+          "kind": "block",
+          "type": "controls_repeat_ext",
+          "values": {
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: Can you change all instances of `"values"` to `'inputs'`?

File: examples/backpack-demo/index.js [hunk 7]
```diff
@@ -7,22 +7,204 @@
...
+            "TIMES": {
+              "block": {
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: Please switch this from `"block"` to `'shadow'`

File: examples/backpack-demo/index.js [hunk 8]
```diff
@@ -7,22 +7,204 @@
...
+              "B": {
+                "shadow": {
+                  "type": "math_number",
+                  "fields": {
+                    "NUM": 1
+                  }
+                }
+              }
+            }
+          },
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: Please fix the indentation here =)

File: examples/backpack-demo/index.js [hunk 9]
```diff
@@ -7,22 +7,204 @@
 /**
  * @fileoverview Backpack demo initialization.
  */
+var toolbox = {
```
Comment (BeksOmega) [inline examples/backpack-demo/index.js]: Can you change this to `const` instead of `var`?

**Review by BeksOmega — CHANGES_REQUESTED** [review]
Generally LGTM! Just a few more fixes and then this should be good to merge =) Thank you for all of your work on this!!

File: examples/backpack-demo/index.js (hunk 9 above)
Comment (varshneydevansh) [reply examples/backpack-demo/index.js]: You are right, `var` should be avoided.

File: examples/backpack-demo/index.js (hunk 5 above)
Comment (varshneydevansh) [reply examples/backpack-demo/index.js]: done

Comment (varshneydevansh) [reply examples/backpack-demo/index.js]: done

**Review by BeksOmega — APPROVED** [review]
Looks wonderful! Thank you so much for the fixes =)

**Pull request merged**